### DISCUSSION TITLE
ngMocks.defaultMock in test.ts not working for MockBuilder + standalone component

### DISCUSSION THREAD

**Discussion by AE1NS** [help-me-mom/ng-mocks#4705] [Q&A] [answered]
I created a reproduction here:
https://codesandbox.io/s/relaxed-tdd-0txsf9?file=/src/test.spec.ts

I declared a ngMocks.defaultMock inside the test.ts but its not working. When you move the code block above the describe inside the test.spec.ts, its working. Am I doing something wrong here or is this a bug?

**Comment by satanTime** [comment, marked as answer]
Hi,

you shouldn't import from spec files.

If you need to define something in the spec file, then defaultMock should be there too.

An example is TitleStrategy above in the code.

**Reply by AE1NS** [reply]
I still dont get that :) I updated the sandbox to show my issue more clearly and replaced the TestService with the TranslateService.
I just followed this one: https://ng-mocks.sudo.eu/api/ngMocks/defaultMock so I placed it inside the test.ts file to be applied to any test. I dont want to define it in each .spec.ts, thats why I wanted to place it only in test.ts.

But as you see, this default mock is not applied to the test and it says, there is 'no provider for translateservice'.

**Reply by satanTime** [reply]
The problem of your example has nothing to do with `ng-mocks`,

your test says that nothing provides `TranslateService`, therefore, you need to define your `TestingModule` in the way the service is provided, for example, to use `TranslateModule.forRoot()`.

```ts
beforeEach(() => MockBuilder(Test1Component).mock(TranslateModule.forRoot()));
```

**Reply by AE1NS** [reply]
Got that point, but my issue is that it doesnt apply the mock from the TranslateService inside the test.ts. I would except a console output of 'Default mock'.

**Reply by satanTime** [reply]
It also works :) but the way codesandbox works, you should define that in `main.ts`, because `test.ts` isn't used in web, only in node.

**Reply by AE1NS** [reply]
Got it. Now it works. Seems that the TranslateModule.forRoot() was my main issue. Many thanks!
Is it also possible do something like `ngMocks.defaultMock(TranslateModule.forRoot());`? Then I wouldnt need to do `.mock(TranslateModule.forRoot())` in each spec file.
